app/controllers/comment_controller.php

```diff
@@ -0,0 +1,64 @@
+<?php
+class CommentController extends AppController
+{
+    function __construct($name)
+    {
+        parent::__construct($name);
+
+        if (!is_logged_in()) 
```

Comment:
Rewritten. Thank you.

*Update*
Removed the constructor for both comment and thread controller and moved it to beforeFilter() method up at app_controller.php. Still used your syntax though, thank you.